File: flutter/flutter-sdk-bridge/ping_core/ios/ping_core/Sources/ping_core/Registry.swift
/*
 * Copyright (c) 2026 Ping Identity Corporation. All rights reserved.
 *
 * This software may be modified and distributed under the terms
 * of the MIT license. See the LICENSE file for details.
 */

import Foundation

/// Lightweight registry for tracking native instances by generated identifiers.
///
/// Implementations generate stable ids, store handles,
/// and clean up when entries are removed.
public protocol Registry: Sendable {
    /// Register a native instance and return the generated id that can be used to retrieve it later.
    func register(_ instance: NativeHandle) async -> String

    /// Look up a previously registered instance by id. Returns `nil` if no match is found.
    func resolve(_ id: String) async -> NativeHandle?

    /// Remove a registered instance by id. Safe to call with unknown ids.
    func remove(_ id: String) async

    /// Remove all registered instances.
    func removeAll() async
}

File: flutter/flutter-sdk-bridge/ping_core/android/src/main/kotlin/com/pingidentity/flutter/core/registry/Registry.kt
/*
 * Copyright (c) 2026 Ping Identity Corporation. All rights reserved.
 *
 * This software may be modified and distributed under the terms
 * of the MIT license. See the LICENSE file for details.
 */

package com.pingidentity.flutter.core.registry

/**
 * A lightweight registry for tracking native instances by generated identifiers.
 *
 * Implementations are responsible for generating stable ids, storing handles,
 * and cleaning up when entries are removed.
 */
interface Registry {
    /**
     * Register a native instance and return the generated id that can be used to
     * retrieve it later.
     */
    fun register(instance: NativeHandle): String

    /**
     * Look up a previously registered instance by id. Returns null if no match is found.
     */
    fun resolve(id: String): NativeHandle?

    /**
     * Remove a registered instance by id. Safe to call with unknown ids.
     */
    fun remove(id: String)

    /**
     * Remove all registered instances.
     */
    fun removeAll()
}
